libs/feature/map/src/lib/utils/map-utils.service.ts

```diff
@@ -128,8 +130,8 @@ export class MapUtilsService {
   /**
    * Will emit `null` if no extent could be computed
    */
-  getLayerExtent(layer: MapContextLayerModel): Observable<Extent | null> {
-    let geographicExtent: Observable<Extent>
+  async getLayerExtent(layer: MapContextLayerModel): Promise<Extent | null> {
```

Comment:
yes! using promises here made everything else so much less verbose, I figured it would be worth it. I usually consider that once "in Angular" we should use observables everywhere, but it was hard not to get the benefit of async functions here.